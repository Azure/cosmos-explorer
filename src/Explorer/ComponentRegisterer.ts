import * as ko from "knockout";
import { DiffEditorComponent } from "./Controls/DiffEditor/DiffEditorComponent";
import { DynamicListComponent } from "./Controls/DynamicList/DynamicListComponent";
import { EditorComponent } from "./Controls/Editor/EditorComponent";
import { JsonEditorComponent } from "./Controls/JsonEditor/JsonEditorComponent";
import { ThroughputInputComponentAutoPilotV3 } from "./Controls/ThroughputInput/ThroughputInputComponentAutoPilotV3";
import * as PaneComponents from "./Panes/PaneComponents";

ko.components.register("editor", new EditorComponent());
ko.components.register("json-editor", new JsonEditorComponent());
ko.components.register("diff-editor", new DiffEditorComponent());
ko.components.register("dynamic-list", DynamicListComponent);
ko.components.register("throughput-input-autopilot-v3", ThroughputInputComponentAutoPilotV3);

// Panes
ko.components.register("cassandra-add-collection-pane", new PaneComponents.CassandraAddCollectionPaneComponent());
