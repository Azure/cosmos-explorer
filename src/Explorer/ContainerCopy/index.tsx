import { MonitorCopyJobsRefState } from "Explorer/ContainerCopy/MonitorCopyJobs/MonitorCopyJobRefState";
import React, { useEffect } from "react";
import CopyJobCommandBar from "./CommandBar/CopyJobCommandBar";
import "./containerCopyStyles.less";
import MonitorCopyJobs, { MonitorCopyJobsRef } from "./MonitorCopyJobs/MonitorCopyJobs";
import { ContainerCopyProps } from "./Types";

const ContainerCopyPanel: React.FC<ContainerCopyProps> = ({ container }) => {
  const monitorCopyJobsRef = React.useRef<MonitorCopyJobsRef>();
  useEffect(() => {
    if (monitorCopyJobsRef.current) {
      MonitorCopyJobsRefState.getState().setRef(monitorCopyJobsRef.current);
    }
  }, [monitorCopyJobsRef.current]);
  return (
    <div id="containerCopyWrapper" className="flexContainer hideOverflows">
      <CopyJobCommandBar container={container} />
      <MonitorCopyJobs ref={monitorCopyJobsRef} />
    </div>
  );
};

export default ContainerCopyPanel;
