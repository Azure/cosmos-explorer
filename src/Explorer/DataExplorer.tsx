import { makeStyles } from "@fluentui/react-components";
import React from "react";
import type { Explorer } from "../Contracts/ViewModels";

interface DataExplorerProps {
  dataExplorer?: Explorer;
}

const useStyles = makeStyles({
  root: {
    backgroundColor: "var(--colorNeutralBackground1)",
    color: "var(--colorNeutralForeground1)",
    height: "100%",
    width: "100%",
  },
});

export const DataExplorer: React.FC<DataExplorerProps> = ({ dataExplorer }) => {
  const styles = useStyles();

  return (
    <div className={`dataExplorerContainer ${styles.root}`}>
      <div>Data Explorer Content</div>
    </div>
  );
};
