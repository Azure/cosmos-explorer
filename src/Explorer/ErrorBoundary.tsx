import React, { Component, ReactNode } from "react";

interface Props {
  children: ReactNode;
}

interface State {
  hasError: boolean;
  error: Error | null;
}

export class ErrorBoundary extends Component<Props, State> {
  public state: State = {
    hasError: false,
    error: null,
  };

  public static getDerivedStateFromError(error: Error): State {
    return { hasError: true, error };
  }

  public componentDidCatch(error: Error): void {
    console.error("Error caught in boundary:", error);
  }

  public render() {
    if (this.state.hasError) {
      return (
        <div style={{ padding: "20px", color: "red" }}>
          <h2>Something went wrong.</h2>
          <details style={{ whiteSpace: "pre-wrap" }}>{this.state.error && this.state.error.toString()}</details>
        </div>
      );
    }

    return this.props.children;
  }
}
