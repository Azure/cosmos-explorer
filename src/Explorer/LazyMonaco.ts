import type * as monaco from "monaco-editor/esm/vs/editor/editor.api";
export type { monaco };

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const loadMonaco = () => import(/* webpackChunkName: "lazy-monaco" */ "monaco-editor/esm/vs/editor/editor.api");
