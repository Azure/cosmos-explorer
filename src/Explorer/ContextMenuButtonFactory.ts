import AddCollectionIcon from "images/AddCollection.svg";
import AddSqlQueryIcon from "images/AddSqlQuery_16x16.svg";
import AddStoredProcedureIcon from "images/AddStoredProcedure.svg";
import AddTriggerIcon from "images/AddTrigger.svg";
import AddUdfIcon from "images/AddUdf.svg";
import DeleteCollectionIcon from "images/DeleteCollection.svg";
import DeleteDatabaseIcon from "images/DeleteDatabase.svg";
import DeleteSprocIcon from "images/DeleteSproc.svg";
import DeleteTriggerIcon from "images/DeleteTrigger.svg";
import DeleteUDFIcon from "images/DeleteUDF.svg";
import HostedTerminalIcon from "images/Hosted-Terminal.svg";
import * as ViewModels from "../Contracts/ViewModels";
import { DefaultAccountExperienceType } from "../DefaultAccountExperienceType";
import { userContext } from "../UserContext";
import { TreeNodeMenuItem } from "./Controls/TreeComponent/TreeComponent";
import Explorer from "./Explorer";
import StoredProcedure from "./Tree/StoredProcedure";
import Trigger from "./Tree/Trigger";
import UserDefinedFunction from "./Tree/UserDefinedFunction";

export interface CollectionContextMenuButtonParams {
  databaseId: string;
  collectionId: string;
}

export interface DatabaseContextMenuButtonParams {
  databaseId: string;
}
/**
 * New resource tree (in ReactJS)
 */
export class ResourceTreeContextMenuButtonFactory {
  public static createDatabaseContextMenu(container: Explorer): TreeNodeMenuItem[] {
    const items: TreeNodeMenuItem[] = [
      {
        iconSrc: AddCollectionIcon,
        onClick: () => container.onNewCollectionClicked(),
        label: container.addCollectionText(),
      },
    ];

    if (userContext.defaultExperience !== DefaultAccountExperienceType.Table) {
      items.push({
        iconSrc: DeleteDatabaseIcon,
        onClick: () => container.openDeleteDatabaseConfirmationPane(),
        label: container.deleteDatabaseText(),
        styleClass: "deleteDatabaseMenuItem",
      });
    }
    return items;
  }

  public static createCollectionContextMenuButton(
    container: Explorer,
    selectedCollection: ViewModels.Collection
  ): TreeNodeMenuItem[] {
    const items: TreeNodeMenuItem[] = [];
    if (userContext.apiType === "SQL" || userContext.apiType === "Gremlin") {
      items.push({
        iconSrc: AddSqlQueryIcon,
        onClick: () => selectedCollection && selectedCollection.onNewQueryClick(selectedCollection, null),
        label: "New SQL Query",
      });
    }

    if (userContext.apiType === "Mongo") {
      items.push({
        iconSrc: AddSqlQueryIcon,
        onClick: () => selectedCollection && selectedCollection.onNewMongoQueryClick(selectedCollection, null),
        label: "New Query",
      });

      items.push({
        iconSrc: HostedTerminalIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewMongoShellClick();
        },
        label: "New Shell",
      });
    }

    if (userContext.apiType === "SQL" || userContext.apiType === "Gremlin") {
      items.push({
        iconSrc: AddStoredProcedureIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewStoredProcedureClick(selectedCollection, null);
        },
        label: "New Stored Procedure",
      });

      items.push({
        iconSrc: AddUdfIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewUserDefinedFunctionClick(selectedCollection, null);
        },
        label: "New UDF",
      });

      items.push({
        iconSrc: AddTriggerIcon,
        onClick: () => {
          const selectedCollection: ViewModels.Collection = container.findSelectedCollection();
          selectedCollection && selectedCollection.onNewTriggerClick(selectedCollection, null);
        },
        label: "New Trigger",
      });
    }

    items.push({
      iconSrc: DeleteCollectionIcon,
      onClick: () => container.openDeleteCollectionConfirmationPane(),
      label: container.deleteCollectionText(),
      styleClass: "deleteCollectionMenuItem",
    });

    return items;
  }

  public static createStoreProcedureContextMenuItems(
    container: Explorer,
    storedProcedure: StoredProcedure
  ): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteSprocIcon,
        onClick: () => storedProcedure.delete(),
        label: "Delete Store Procedure",
      },
    ];
  }

  public static createTriggerContextMenuItems(container: Explorer, trigger: Trigger): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteTriggerIcon,
        onClick: () => trigger.delete(),
        label: "Delete Trigger",
      },
    ];
  }

  public static createUserDefinedFunctionContextMenuItems(
    container: Explorer,
    userDefinedFunction: UserDefinedFunction
  ): TreeNodeMenuItem[] {
    if (userContext.apiType === "Cassandra") {
      return [];
    }

    return [
      {
        iconSrc: DeleteUDFIcon,
        onClick: () => userDefinedFunction.delete(),
        label: "Delete User Defined Function",
      },
    ];
  }
}
