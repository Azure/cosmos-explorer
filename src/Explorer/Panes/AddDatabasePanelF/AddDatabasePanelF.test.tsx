import { shallow } from "enzyme";
import React from "react";
import Explorer from "../../Explorer";
import { AddDatabasePanelF } from "./AddDatabasePanelF";

const props = {
  explorer: new Explorer(),
  closePanel: (): void => undefined,
  openNotificationConsole: (): void => undefined,
};

describe("AddDatabasePane Pane", () => {
  it("should render Default properly", () => {
    const wrapper = shallow(<AddDatabasePanelF {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
