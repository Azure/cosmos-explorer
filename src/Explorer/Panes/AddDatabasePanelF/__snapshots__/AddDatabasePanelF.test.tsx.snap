// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDatabasePane Pane should render Default properly 1`] = `
<GenericRightPaneComponent
  container={
    Explorer {
      "_closeModalDialog": [Function],
      "_closeSynapseLinkModalDialog": [Function],
      "_isAfecFeatureRegistered": [Function],
      "_isInitializingNotebooks": false,
      "_panes": Array [
        AddDatabasePane {
          "autoPilotUsageCost": [Function],
          "canConfigureThroughput": [Function],
          "canExceedMaximumValue": [Function],
          "canRequestSupport": [Function],
          "container": [Circular],
          "costsVisible": [Function],
          "databaseCreateNewShared": [Function],
          "databaseId": [Function],
          "databaseIdLabel": [Function],
          "databaseIdPlaceHolder": [Function],
          "databaseIdTooltipText": [Function],
          "databaseLevelThroughputTooltipText": [Function],
          "firstFieldHasFocus": [Function],
          "formErrors": [Function],
          "formErrorsDetails": [Function],
          "freeTierExceedThroughputTooltip": [Function],
          "id": "adddatabasepane",
          "isAutoPilotSelected": [Function],
          "isExecuting": [Function],
          "isFreeTierAccount": [Function],
          "isTemplateReady": [Function],
          "maxAutoPilotThroughputSet": [Function],
          "maxThroughputRU": [Function],
          "maxThroughputRUText": [Function],
          "minThroughputRU": [Function],
          "onMoreDetailsKeyPress": [Function],
          "requestUnitsUsageCost": [Function],
          "ruToolTipText": [Function],
          "showUpsellMessage": [Function],
          "throughput": [Function],
          "throughputRangeText": [Function],
          "throughputSpendAck": [Function],
          "throughputSpendAckText": [Function],
          "throughputSpendAckVisible": [Function],
          "title": [Function],
          "upsellAnchorText": [Function],
          "upsellAnchorUrl": [Function],
          "upsellMessage": [Function],
          "upsellMessageAriaLabel": [Function],
          "visible": [Function],
        },
        AddCollectionPane {
          "_isSynapseLinkEnabled": [Function],
          "autoPilotThroughput": [Function],
          "autoPilotUsageCost": [Function],
          "canConfigureThroughput": [Function],
          "canExceedMaximumValue": [Function],
          "canRequestSupport": [Function],
          "collectionId": [Function],
          "collectionIdTitle": [Function],
          "collectionWithThroughputInShared": [Function],
          "collectionWithThroughputInSharedTitle": [Function],
          "container": [Circular],
          "costsVisible": [Function],
          "databaseCreateNew": [Function],
          "databaseCreateNewShared": [Function],
          "databaseHasSharedOffer": [Function],
          "databaseId": [Function],
          "databaseIds": [Function],
          "dedicatedRequestUnitsUsageCost": [Function],
          "displayCollectionThroughput": [Function],
          "firstFieldHasFocus": [Function],
          "formErrors": [Function],
          "formErrorsDetails": [Function],
          "formWarnings": [Function],
          "freeTierExceedThroughputTooltip": [Function],
          "id": "addcollectionpane",
          "isAnalyticalStorageOn": [Function],
          "isAutoPilotSelected": [Function],
          "isExecuting": [Function],
          "isFixedStorageSelected": [Function],
          "isFreeTierAccount": [Function],
          "isNonTableApi": [Function],
          "isPreferredApiTable": [Function],
          "isSharedAutoPilotSelected": [Function],
          "isSynapseLinkSupported": [Function],
          "isSynapseLinkUpdating": [Function],
          "isTemplateReady": [Function],
          "isTryCosmosDBSubscription": [Function],
          "isUnlimitedStorageSelected": [Function],
          "largePartitionKey": [Function],
          "lowerCasePartitionKeyName": [Function],
          "maxCollectionsReached": [Function],
          "maxCollectionsReachedMessage": [Function],
          "maxThroughputRU": [Function],
          "minThroughputRU": [Function],
          "onMoreDetailsKeyPress": [Function],
          "partitionKey": [Function],
          "partitionKeyName": [Function],
          "partitionKeyPattern": [Function],
          "partitionKeyPlaceholder": [Function],
          "partitionKeyTitle": [Function],
          "partitionKeyVisible": [Function],
          "requestUnitsUsageCost": [Function],
          "ruToolTipText": [Function],
          "sharedAutoPilotThroughput": [Function],
          "sharedThroughputRangeText": [Function],
          "shouldCreateMongoWildcardIndex": [Function],
          "shouldUseDatabaseThroughput": [Function],
          "showAnalyticalStore": [Function],
          "showEnableSynapseLink": [Function],
          "showIndexingOptionsForSharedThroughput": [Function],
          "showUpsellMessage": [Function],
          "storage": [Function],
          "throughputDatabase": [Function],
          "throughputMultiPartition": [Function],
          "throughputRangeText": [Function],
          "throughputSinglePartition": [Function],
          "throughputSpendAck": [Function],
          "throughputSpendAckText": [Function],
          "throughputSpendAckVisible": [Function],
          "title": [Function],
          "ttl90DaysEnabled": [Function],
          "uniqueKeys": [Function],
          "uniqueKeysPlaceholder": [Function],
          "uniqueKeysVisible": [Function],
          "upsellAnchorText": [Function],
          "upsellAnchorUrl": [Function],
          "upsellMessage": [Function],
          "upsellMessageAriaLabel": [Function],
          "useIndexingForSharedThroughput": [Function],
          "visible": [Function],
        },
        GraphStylingPane {
          "container": [Circular],
          "firstFieldHasFocus": [Function],
          "formErrors": [Function],
          "formErrorsDetails": [Function],
          "graphConfigUIData": Object {
            "nodeCaptionChoice": [Function],
            "nodeColorKeyChoice": [Function],
            "nodeIconChoice": [Function],
            "nodeIconSet": [Function],
            "nodeProperties": [Function],
            "nodePropertiesWithNone": [Function],
            "showNeighborType": [Function],
          },
          "id": "graphstylingpane",
          "isExecuting": [Function],
          "isTemplateReady": [Function],
          "title": [Function],
          "visible": [Function],
        },
        CassandraAddCollectionPane {
          "autoPilotUsageCost": [Function],
          "canConfigureThroughput": [Function],
          "canExceedMaximumValue": [Function],
          "canRequestSupport": [Function],
          "container": [Circular],
          "costsVisible": [Function],
          "createTableQuery": [Function],
          "dedicateTableThroughput": [Function],
          "firstFieldHasFocus": [Function],
          "formErrors": [Function],
          "formErrorsDetails": [Function],
          "id": "cassandraaddcollectionpane",
          "isAutoPilotSelected": [Function],
          "isExecuting": [Function],
          "isFreeTierAccount": [Function],
          "isSharedAutoPilotSelected": [Function],
          "isTemplateReady": [Function],
          "keyspaceCreateNew": [Function],
          "keyspaceHasSharedOffer": [Function],
          "keyspaceId": [Function],
          "keyspaceIds": [Function],
          "keyspaceOffers": Map {},
          "keyspaceThroughput": [Function],
          "maxThroughputRU": [Function],
          "minThroughputRU": [Function],
          "requestUnitsUsageCostDedicated": [Function],
          "requestUnitsUsageCostShared": [Function],
          "ruToolTipText": [Function],
          "selectedAutoPilotThroughput": [Function],
          "sharedAutoPilotThroughput": [Function],
          "sharedThroughputRangeText": [Function],
          "sharedThroughputSpendAck": [Function],
          "sharedThroughputSpendAckText": [Function],
          "sharedThroughputSpendAckVisible": [Function],
          "tableId": [Function],
          "throughput": [Function],
          "throughputRangeText": [Function],
          "throughputSpendAck": [Function],
          "throughputSpendAckText": [Function],
          "throughputSpendAckVisible": [Function],
          "title": [Function],
          "userTableQuery": [Function],
          "visible": [Function],
        },
      ],
      "_refreshSparkEnabledStateForAccount": [Function],
      "_resetNotebookWorkspace": [Function],
      "addCollectionPane": AddCollectionPane {
        "_isSynapseLinkEnabled": [Function],
        "autoPilotThroughput": [Function],
        "autoPilotUsageCost": [Function],
        "canConfigureThroughput": [Function],
        "canExceedMaximumValue": [Function],
        "canRequestSupport": [Function],
        "collectionId": [Function],
        "collectionIdTitle": [Function],
        "collectionWithThroughputInShared": [Function],
        "collectionWithThroughputInSharedTitle": [Function],
        "container": [Circular],
        "costsVisible": [Function],
        "databaseCreateNew": [Function],
        "databaseCreateNewShared": [Function],
        "databaseHasSharedOffer": [Function],
        "databaseId": [Function],
        "databaseIds": [Function],
        "dedicatedRequestUnitsUsageCost": [Function],
        "displayCollectionThroughput": [Function],
        "firstFieldHasFocus": [Function],
        "formErrors": [Function],
        "formErrorsDetails": [Function],
        "formWarnings": [Function],
        "freeTierExceedThroughputTooltip": [Function],
        "id": "addcollectionpane",
        "isAnalyticalStorageOn": [Function],
        "isAutoPilotSelected": [Function],
        "isExecuting": [Function],
        "isFixedStorageSelected": [Function],
        "isFreeTierAccount": [Function],
        "isNonTableApi": [Function],
        "isPreferredApiTable": [Function],
        "isSharedAutoPilotSelected": [Function],
        "isSynapseLinkSupported": [Function],
        "isSynapseLinkUpdating": [Function],
        "isTemplateReady": [Function],
        "isTryCosmosDBSubscription": [Function],
        "isUnlimitedStorageSelected": [Function],
        "largePartitionKey": [Function],
        "lowerCasePartitionKeyName": [Function],
        "maxCollectionsReached": [Function],
        "maxCollectionsReachedMessage": [Function],
        "maxThroughputRU": [Function],
        "minThroughputRU": [Function],
        "onMoreDetailsKeyPress": [Function],
        "partitionKey": [Function],
        "partitionKeyName": [Function],
        "partitionKeyPattern": [Function],
        "partitionKeyPlaceholder": [Function],
        "partitionKeyTitle": [Function],
        "partitionKeyVisible": [Function],
        "requestUnitsUsageCost": [Function],
        "ruToolTipText": [Function],
        "sharedAutoPilotThroughput": [Function],
        "sharedThroughputRangeText": [Function],
        "shouldCreateMongoWildcardIndex": [Function],
        "shouldUseDatabaseThroughput": [Function],
        "showAnalyticalStore": [Function],
        "showEnableSynapseLink": [Function],
        "showIndexingOptionsForSharedThroughput": [Function],
        "showUpsellMessage": [Function],
        "storage": [Function],
        "throughputDatabase": [Function],
        "throughputMultiPartition": [Function],
        "throughputRangeText": [Function],
        "throughputSinglePartition": [Function],
        "throughputSpendAck": [Function],
        "throughputSpendAckText": [Function],
        "throughputSpendAckVisible": [Function],
        "title": [Function],
        "ttl90DaysEnabled": [Function],
        "uniqueKeys": [Function],
        "uniqueKeysPlaceholder": [Function],
        "uniqueKeysVisible": [Function],
        "upsellAnchorText": [Function],
        "upsellAnchorUrl": [Function],
        "upsellMessage": [Function],
        "upsellMessageAriaLabel": [Function],
        "useIndexingForSharedThroughput": [Function],
        "visible": [Function],
      },
      "addCollectionText": [Function],
      "addDatabasePane": AddDatabasePane {
        "autoPilotUsageCost": [Function],
        "canConfigureThroughput": [Function],
        "canExceedMaximumValue": [Function],
        "canRequestSupport": [Function],
        "container": [Circular],
        "costsVisible": [Function],
        "databaseCreateNewShared": [Function],
        "databaseId": [Function],
        "databaseIdLabel": [Function],
        "databaseIdPlaceHolder": [Function],
        "databaseIdTooltipText": [Function],
        "databaseLevelThroughputTooltipText": [Function],
        "firstFieldHasFocus": [Function],
        "formErrors": [Function],
        "formErrorsDetails": [Function],
        "freeTierExceedThroughputTooltip": [Function],
        "id": "adddatabasepane",
        "isAutoPilotSelected": [Function],
        "isExecuting": [Function],
        "isFreeTierAccount": [Function],
        "isTemplateReady": [Function],
        "maxAutoPilotThroughputSet": [Function],
        "maxThroughputRU": [Function],
        "maxThroughputRUText": [Function],
        "minThroughputRU": [Function],
        "onMoreDetailsKeyPress": [Function],
        "requestUnitsUsageCost": [Function],
        "ruToolTipText": [Function],
        "showUpsellMessage": [Function],
        "throughput": [Function],
        "throughputRangeText": [Function],
        "throughputSpendAck": [Function],
        "throughputSpendAckText": [Function],
        "throughputSpendAckVisible": [Function],
        "title": [Function],
        "upsellAnchorText": [Function],
        "upsellAnchorUrl": [Function],
        "upsellMessage": [Function],
        "upsellMessageAriaLabel": [Function],
        "visible": [Function],
      },
      "addDatabaseText": [Function],
      "arcadiaToken": [Function],
      "canExceedMaximumValue": [Function],
      "canSaveQueries": [Function],
      "cassandraAddCollectionPane": CassandraAddCollectionPane {
        "autoPilotUsageCost": [Function],
        "canConfigureThroughput": [Function],
        "canExceedMaximumValue": [Function],
        "canRequestSupport": [Function],
        "container": [Circular],
        "costsVisible": [Function],
        "createTableQuery": [Function],
        "dedicateTableThroughput": [Function],
        "firstFieldHasFocus": [Function],
        "formErrors": [Function],
        "formErrorsDetails": [Function],
        "id": "cassandraaddcollectionpane",
        "isAutoPilotSelected": [Function],
        "isExecuting": [Function],
        "isFreeTierAccount": [Function],
        "isSharedAutoPilotSelected": [Function],
        "isTemplateReady": [Function],
        "keyspaceCreateNew": [Function],
        "keyspaceHasSharedOffer": [Function],
        "keyspaceId": [Function],
        "keyspaceIds": [Function],
        "keyspaceOffers": Map {},
        "keyspaceThroughput": [Function],
        "maxThroughputRU": [Function],
        "minThroughputRU": [Function],
        "requestUnitsUsageCostDedicated": [Function],
        "requestUnitsUsageCostShared": [Function],
        "ruToolTipText": [Function],
        "selectedAutoPilotThroughput": [Function],
        "sharedAutoPilotThroughput": [Function],
        "sharedThroughputRangeText": [Function],
        "sharedThroughputSpendAck": [Function],
        "sharedThroughputSpendAckText": [Function],
        "sharedThroughputSpendAckVisible": [Function],
        "tableId": [Function],
        "throughput": [Function],
        "throughputRangeText": [Function],
        "throughputSpendAck": [Function],
        "throughputSpendAckText": [Function],
        "throughputSpendAckVisible": [Function],
        "title": [Function],
        "userTableQuery": [Function],
        "visible": [Function],
      },
      "clickHostedAccountSwitch": [Function],
      "clickHostedDirectorySwitch": [Function],
      "closeDialog": undefined,
      "closeSidePanel": undefined,
      "collapsedResourceTreeWidth": 36,
      "collectionCreationDefaults": Object {
        "storage": "100",
        "throughput": Object {
          "fixed": 400,
          "shared": 400,
          "unlimited": 400,
          "unlimitedmax": 1000000,
          "unlimitedmin": 400,
        },
      },
      "collectionTitle": [Function],
      "collectionTreeNodeAltText": [Function],
      "commandBarComponentAdapter": CommandBarComponentAdapter {
        "container": [Circular],
        "isNotebookTabActive": [Function],
        "parameters": [Function],
        "tabsButtons": Array [],
      },
      "databaseAccount": [Function],
      "databases": [Function],
      "defaultExperience": [Function],
      "deleteCollectionText": [Function],
      "deleteDatabaseText": [Function],
      "graphStylingPane": GraphStylingPane {
        "container": [Circular],
        "firstFieldHasFocus": [Function],
        "formErrors": [Function],
        "formErrorsDetails": [Function],
        "graphConfigUIData": Object {
          "nodeCaptionChoice": [Function],
          "nodeColorKeyChoice": [Function],
          "nodeIconChoice": [Function],
          "nodeIconSet": [Function],
          "nodeProperties": [Function],
          "nodePropertiesWithNone": [Function],
          "showNeighborType": [Function],
        },
        "id": "graphstylingpane",
        "isExecuting": [Function],
        "isTemplateReady": [Function],
        "title": [Function],
        "visible": [Function],
      },
      "hasStorageAnalyticsAfecFeature": [Function],
      "isAccountReady": [Function],
      "isAutoscaleDefaultEnabled": [Function],
      "isEnableMongoCapabilityPresent": [Function],
      "isFixedCollectionWithSharedThroughputSupported": [Function],
      "isGitHubPaneEnabled": [Function],
      "isHostedDataExplorerEnabled": [Function],
      "isLeftPaneExpanded": [Function],
      "isMongoIndexingEnabled": [Function],
      "isNotebookEnabled": [Function],
      "isNotebooksEnabledForAccount": [Function],
      "isPublishNotebookPaneEnabled": [Function],
      "isResourceTokenCollectionNodeSelected": [Function],
      "isRightPanelV2Enabled": [Function],
      "isSchemaEnabled": [Function],
      "isServerlessEnabled": [Function],
      "isSparkEnabled": [Function],
      "isSparkEnabledForAccount": [Function],
      "isSynapseLinkUpdating": [Function],
      "isTabsContentExpanded": [Function],
      "memoryUsageInfo": [Function],
      "notebookBasePath": [Function],
      "notebookServerInfo": [Function],
      "onRefreshDatabasesKeyPress": [Function],
      "onRefreshResourcesClick": [Function],
      "onSwitchToConnectionString": [Function],
      "openDialog": undefined,
      "openSidePanel": undefined,
      "provideFeedbackEmail": [Function],
      "queriesClient": QueriesClient {
        "container": [Circular],
      },
      "refreshDatabaseAccount": [Function],
      "refreshNotebookList": [Function],
      "refreshTreeTitle": [Function],
      "resourceTokenCollection": [Function],
      "resourceTokenCollectionId": [Function],
      "resourceTokenDatabaseId": [Function],
      "resourceTokenPartitionKey": [Function],
      "resourceTree": ResourceTreeAdapter {
        "container": [Circular],
        "copyNotebook": [Function],
        "databaseCollectionIdMap": Map {},
        "koSubsCollectionIdMap": Map {},
        "koSubsDatabaseIdMap": Map {},
        "parameters": [Function],
      },
      "resourceTreeForResourceToken": ResourceTreeAdapterForResourceToken {
        "container": [Circular],
        "parameters": [Function],
      },
      "selectedDatabaseId": [Function],
      "selectedNode": [Function],
      "setInProgressConsoleDataIdToBeDeleted": undefined,
      "setIsNotificationConsoleExpanded": undefined,
      "setNotificationConsoleData": undefined,
      "signInAad": [Function],
      "sparkClusterConnectionInfo": [Function],
      "splitter": Splitter {
        "bounds": Object {
          "max": 400,
          "min": 240,
        },
        "direction": "vertical",
        "isCollapsed": [Function],
        "leftSideId": "resourcetree",
        "onResizeStart": [Function],
        "onResizeStop": [Function],
        "splitterId": "h_splitter1",
      },
      "tabsManager": TabsManager {
        "activeTab": [Function],
        "openedTabs": [Function],
      },
      "toggleLeftPaneExpandedKeyPress": [Function],
    }
  }
  formError=""
  id="add-database-inputs"
  isExecuting={false}
  onClose={[Function]}
  onSubmit={[Function]}
  submitButtonText="OK"
  title="New Database"
>
  <form
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      aria-labelledby="databaseTitle"
      className="paneContentContainer"
      role="dialog"
    >
      <PanelInfoErrorComponent
        link="https://aka.ms/azure-cosmos-db-pricing"
        linkText="More details"
        message="Start at $24/mo per database, multiple containers included"
        messageType="info"
        openNotificationConsole={[Function]}
        showErrorDetails={false}
      />
      <div
        className="paneMainContent"
      >
        <div>
          <p>
            <span
              className="mandatoryStar"
            >
              *
            </span>
            <Text
              variant="small"
            >
              Database id
            </Text>
            <Tooltip>
              A database is a logical container of one or more collections
            </Tooltip>
          </p>
          <StyledTextFieldBase
            aria-label="Database id"
            aria-required="true"
            autoComplete="off"
            autoFocus={true}
            id="database-id"
            onChange={[Function]}
            pattern="[^/?#\\\\\\\\]*[^/?# \\\\\\\\]"
            placeholder="Type a new database id"
            size={40}
            title="May not end with space nor contain characters '\\\\' '/' '#' '?'"
            type="text"
            value=""
          />
          <div
            aria-label="New database provision support"
            className="databaseProvision"
            style={
              Object {
                "display": "block ruby",
              }
            }
          >
            <StyledCheckboxBase
              checked={true}
              label="Provision throughput"
              onChange={[Function]}
              styles={
                Object {
                  "checkbox": Object {
                    "height": 12,
                    "width": 12,
                  },
                  "label": Object {
                    "alignItems": "center",
                    "padding": 0,
                  },
                }
              }
              tabIndex={0}
              title="Provision shared throughput"
            />
             
            <Tooltip>
              Provisioned throughput at the database level will be shared across all collections within the database.
            </Tooltip>
          </div>
          <div>
            <ThroughputInput
              isAutoscaleSelected={false}
              isDatabase={true}
              onCostAcknowledgeChange={[Function]}
              setIsAutoscale={[Function]}
              setThroughputValue={[Function]}
              throughput={400}
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</GenericRightPaneComponent>
`;
